### PR TITLE
feat: initial support for prettier-stylelint

### PR DESCRIPTION
**Pull request by hugomrdias**
this adds initial support for [prettier-stylelint](https://github.com/hugomrdias/prettier-stylelint), im still adding features but the basic stuff at least for the extension works well.

please keep in mind that i don't do ts or vscode extensions much :) be gentle 

ref #207

### CODE REVIEW

**Comment by CiGit** [comment]
Thanks for that PR and your prettier-stylelint.
Could your prettier-stylint format be synchronous ? To be able to use `safeExecution` without having to export everything ?

**Comment by hugomrdias** [comment]
stylelint doesnt have a sync api, i can create a safeExecutionAsync instead and leave everything abstracted in the utils

**Comment by CiGit** [comment]
This is also a solution. Make `safeExecution` handle both.

**Comment by hugomrdias** [comment]
@CiGit does this seem good to you ?

**Comment by CiGit** [comment]
Yup, I prefer. asking @RobinMalfait  for a review of `safeExecution`

About prettier-stylelint, which prettier/ stylelint version does it pickup?

File: src/errorHandler.ts [hunk 1]
```diff
@@ -110,10 +110,24 @@ export function addToOutput(message: string): void {
...
+        return cb
+            .then(returnValue => {
+                updateStatusBar('Prettier: $(check)');
+                return returnValue;
+            })
+            .catch((err: Error) => {
+                addToOutput(addFilePath(err.message, fileName));
+                updateStatusBar('Prettier: $(x)');
+
+                console.log(err);
```
Comment (RobinMalfait) [inline src/errorHandler.ts]: Maybe we need to remove this line

Comment (RobinMalfait) [reply src/errorHandler.ts]: Hmm, now that I'm thinking about it, the `safeExecution` has a lot of duplicated code.

Comment (hugomrdias) [reply src/errorHandler.ts]: Ofc damn .. I'll remove that

**Comment by hugomrdias** [comment]
@cigit for now the bundled version, but I'll be fixing that in a near future

**Comment by hugomrdias** [comment]
@RobinMalfait pushed a commit removing the console log

**Comment by CiGit** [comment]
My concerns are how often do you expect it to be updated, until it has all the feature you'd like ?
On your repo I see 9 releases in 8 days. I don't think we will be able to follow so many releases here :-)

**Comment by hugomrdias** [comment]
i dont expect to have that many releases from now on. besides ill keep sending PR's here for each new release.
plus after i add support for relative package lookup (which im already working on) this extension need to have the latest version bundled to work properly

**Comment by hugomrdias** [comment]
@CiGit with this last commit your concerns should be addressed

**Comment by CiGit** [comment]
Please resolve conflicts, I'll merge after.

**Comment by CiGit** [comment]
And update the readme please :-)

**Comment by CiGit** [comment]
Thanks !

**Comment by WickyNilliams** [comment]
Is there some way to configure this to run over JS files? Use case: styled-components.

The regular stylelint extension for vscode has a config option `additionalDocumentSelectors` so you can configure it to run over JS files as well as the usuall css/scss/less. eg:

```js
  "stylelint.additionalDocumentSelectors": [
    "javascriptreact",
    "javascript"
  ]
```

Happy to make an issue in the appropriate repo, but thought i'd ask here given this was the integration point

**Pull request merged**